packages/datagateway-download/src/downloadCart/downloadCartTable.component.tsx

```diff
@@ -351,51 +352,83 @@ const DownloadCartTable: React.FC<DownloadCartTableProps> = (
         >
           <Grid
             container
-            item
+            spacing={1}
             justify="flex-end"
             alignItems="center"
             direction="row"
             style={{ marginRight: '1.2em' }}
           >
-            {fileCountsLoading && (
-              <CircularProgress
-                size={15}
-                thickness={7}
-                disableShrink={true}
-                aria-label={t('downloadCart.calculating')}
-              />
+            <Grid item>
+              {fileCountsLoading && (
+                <CircularProgress
+                  size={15}
+                  thickness={7}
+                  disableShrink={true}
+                  aria-label={t('downloadCart.calculating')}
+                />
+              )}
+              <Typography id="fileCountDisplay" style={{ marginLeft: '4px' }}>
+                {t('downloadCart.number_of_files')}:{' '}
+                {fileCount !== -1
+                  ? fileCount
+                  : `${t('downloadCart.calculating')}...`}
+                {fileCountMax !== -1 && ` / ${fileCountMax}`}
+              </Typography>
+            </Grid>
+            {fileCount > fileCountMax && (
+              <Grid item>
+                <Alert
+                  id="fileLimitAlert"
+                  variant="filled"
+                  severity="error"
+                  icon={false}
+                  style={{ padding: '0px 8px', lineHeight: 0.6 }}
+                >
+                  Too many files - you have exceeded limit of {fileCountMax}{' '}
+                  files - please remove some files
```

Comment:
These strings need to be in the translation file - you should still be able to parameterise `fileCountMax` and `totalSizeMax`

Reply:
I've given this some thought and decided to combine our ideas. I say we provide one `<Alert>` object, like the one I've done, in the place where your `empty_items_warning` is so that they aren't too many error messages on the page. We keep that `<Alert>` there, with the appropriate error message, until no further errors remain on the page.